src/solver/optimisation/opt_construction_variables_optimisees_lineaire.cpp

```diff
@@ -104,11 +105,22 @@ void OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeLineaire(PROBLEME_HEBD
                   = VARIABLE_BORNEE_DES_DEUX_COTES;
                 NombreDeVariables++;
                 // 3. Level
-                CorrespondanceVarNativesVarOptim->ShortTermStorage.LevelVariable[globalIndex]
-                  = NombreDeVariables;
-                ProblemeAResoudre->TypeDeVariable[NombreDeVariables]
-                  = VARIABLE_BORNEE_DES_DEUX_COTES;
-                NombreDeVariables++;
+                if (pdt == 0 || (pdt % storage.storagecycle != 0))
```

Comment:
There is something I don't understand here : 
If we assume that we don't yet handle the case **initiallevel = optim** (as suggests the code reading the file **list.ini**), the initial level should not be a variable of the optimization, but should show up in the some constraint RHS.
Why then do we create a new optimization variable when **pdt == 0**, as suggests the block  `if (pdt == 0 || (pdt % storage.storagecycle != 0))` ?

For me, there only should be (still assuming that case **initiallevel = optim** is not handled) : 
```C++
if (pdt % storage.storagecycle != 0)
{
   ...
}
```
and that's it. For me, no **else** required.